rpc/client.go

```diff
@@ -53,18 +54,30 @@ type Client struct {
 // reads from this goroutine.
 func (c *Client) readPump(funcMap map[string]eth.RPCFunc, logger log.TMLogger) {
 	defer func() {
+		if r := recover(); r != nil {
+			logger.Error("WebSocket read panicked", "err", r)
+		}
 		c.hub.unregister <- c
-		_ = c.conn.Close()
+		if err := c.conn.Close(); err != nil {
+			logger.Error("Error closing WebSocket read pump", "err", err)
+		}
 	}()
 	c.conn.SetReadLimit(maxMessageSize)
 	_ = c.conn.SetReadDeadline(time.Now().Add(pongWait))
 	c.conn.SetPongHandler(func(string) error { _ = c.conn.SetReadDeadline(time.Now().Add(pongWait)); return nil })
 	for {
 		_, message, err := c.conn.ReadMessage()
 		if err != nil {
-			logger.Error("websocket client read message error", "err", err)
-			websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure)
-			break
+			if websocket.IsUnexpectedCloseError(
+				err,
+				websocket.CloseGoingAway,
+				websocket.CloseAbnormalClosure,
+			) {
+				logger.Error("WebSocket unexpected close error", "err", err)
+			} else {
+				logger.Debug("WebSocket close request", "err", err)
+			}
+			return
```

Comment:
There was a `break` here before, now there's a `return`, was this change intentional?

Reply:
No reason that I remember, both should do the same thing. I guess `return` is more explicit.